File: c++/two-sum-ii-input-array-is-sorted.cc
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;
        while (l < r)
        {
            if (numbers.at(l) + numbers.at(r) > target)
                r--;
            else if (numbers.at(l) + numbers.at(r) < target)
                l++;
            else
                return {l + 1, r + 1};
        }
        return {0};
    }
};

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        unordered_map<int, int> hashmap;
        for (int i = 0; i < numbers.size(); i++) {
            auto it = hashmap.find(numbers[i]);
            if (it != hashmap.end()) {
                return {it->second + 1, i + 1};
            }
            hashmap[target - numbers[i]] = i;
        }
        return {};
    }
};

File: python/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        leftPtr = 0
        rightPtr = len(numbers) - 1

        while (rightPtr > leftPtr):
            total = numbers[leftPtr] + numbers[rightPtr]
            if total < target:
                leftPtr += 1
            elif total > target:
                rightPtr -= 1
            else:
                return [leftPtr + 1, rightPtr + 1]

        return []
